packages/cactus-plugin-ledger-connector-quorum/src/test/typescript/integration/plugin-ledger-connector-quorum/deploy-contract/v2.3.0-deploy-contract-from-json-private.test.ts

```diff
@@ -0,0 +1,242 @@
+  pluginRegistry2.add(connector2);
+
+  const connector3 = await pluginFactoryLedgerConnector.create({
+    instanceId: connectorInstanceId3,
+    pluginRegistry: pluginRegistry3,
+    rpcApiHttpHost: rpcApiHttpHostMember3,
+    rpcApiWsHost: rpcApiWsHostMember3,
+    logLevel,
+  });
+  t.ok(connector3, "connector3 truthy OK");
+  test.onFinish(() => connector3.shutdown());
+  pluginRegistry3.add(connector3);
+
+  await connector1.onPluginInit();
+  await connector2.onPluginInit();
+  await connector3.onPluginInit();
+
+  const chainIdMember1 = await web3Member1.eth.getChainId();
+  t.comment(`chainIdMember1=${chainIdMember1}`);
+  const chainIdMember2 = await web3Member2.eth.getChainId();
+  t.comment(`chainIdMember2=${chainIdMember2}`);
+  const chainIdMember3 = await web3Member3.eth.getChainId();
+  t.comment(`chainIdMember3=${chainIdMember3}`);
+
+  const web3QuorumMember1 = Web3JsQuorum(web3Member1);
+  t.ok(web3QuorumMember1, "web3QuorumMember1 truthy OK");
+
+  const web3QuorumMember2 = Web3JsQuorum(web3Member2);
+  t.ok(web3QuorumMember2, "web3QuorumMember2 truthy OK");
+
+  const web3QuorumMember3 = Web3JsQuorum(web3Member3);
+  t.ok(web3QuorumMember3, "web3QuorumMember3 truthy OK");
+
+  const accountAddress = keys.quorum.member1.accountAddress;
+  const deployRes = await connector1.deployContract({
+    contractName: HelloWorldContractJson.contractName,
+    web3SigningCredential: {
+      secret: keys.quorum.member1.privateKey,
+      type: Web3SigningCredentialType.PrivateKeyHex,
+      ethAccount: accountAddress,
+    },
+    keychainId: keychainId1,
+    gas: 3000000,
+    gasPrice: 0,
+  });
+
+  t.ok(deployRes, "deployRes truthy OK");
+  t.ok(deployRes.transactionReceipt, "deployRes.transactionReceipt truthy OK");
+  t.ok(
+    deployRes.transactionReceipt.contractAddress,
+    "deployRes.transactionReceipt.contractAddress truthy OK",
+  );
+
+  const contractDeployReceipt = await web3QuorumMember1.eth.getTransactionReceipt(
+    deployRes.transactionReceipt.transactionHash,
+  );
+  t.ok(contractDeployReceipt, "contractDeployReceipt truthy OK");
+  const receipt = contractDeployReceipt;
+  const { contractAddress } = receipt;
+  t.ok(contractAddress, "contractAddress truthy OK");
+
+  const member1Contract = new web3QuorumMember1.eth.Contract(
+    HelloWorldContractJson.abi as AbiItem[],
+    contractAddress,
+  );
+  const mem1Response = await member1Contract.methods.getName().call();
+  t.strictEqual(mem1Response, "CaptainCactus");
+
+  const member2Contract = new web3QuorumMember2.eth.Contract(
+    HelloWorldContractJson.abi as AbiItem[],
+    contractAddress,
+  );
+  const mem2Response = await member2Contract.methods.getName().call();
+  t.strictEqual(mem2Response, "CaptainCactus");
+
+  const member3Contract = new web3QuorumMember3.eth.Contract(
+    HelloWorldContractJson.abi as AbiItem[],
+    contractAddress,
+  );
+  await t.doesNotReject(member3Contract.methods.getName().call);
```

Comment:
@aldousalvarez @jagpreetsinghsasan @johnhomantaring This test looks like it just deploys a public contract and then does a public `CALL` on it. No actual transaction is executed that would modify private state (`SEND` op).

Could you please make this test case a little more thorough so that:
1. It deploys a private contract that only 2 members out of 3 can access
2. It verifies that when the third member is trying to access the contract from point 1): it fails (assert the fact of the rejection of the promise)
3. It verifies that member 1 and 2 can indeed run `SEND` transactions privately to the contract. (You can update the name in the contract instance variable to some UUID and then verify that member 1 and 2 can read that and member 3 cannot (either with asserting complete rejection or by observing that the name variable for member 3 is still the old/original/default name not the UUID that was set by the other members)
4. Make the tests cases use the connector itself not the `web3js-quorum` library because otherwise we are just testing that the `web3js-quorum` library itself works correctly not that the connector plugin that uses it under its own hood.

Reply:
Yes, the test case should be changed to incorporate the testing of the connector class itself via connector class methods related to private transactions.

Reply:
The new test case looks good, thank you all!